### ISSUE TITLE
[Modding] Support for NPC merchants is really bad

### ISSUE THREAD

**Issue by ubertpendragon** [Regalis11/Barotrauma#13233]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The way NPC merchant stores are generated is really cumbersome for modding. 

Where items are sold is defined directly inside the item itself, which means if you want an item to be sold in a custom merchant NPC, you need to override each individual item you want to be sold there which causes awful amounts of incompatibility.

Modders resort to vending machines to circumvent the problem entirely.

![image](https://github.com/Regalis11/Barotrauma/assets/73229309/5e57e416-7bcf-4713-bdb0-e726abea94e3)
![image](https://github.com/Regalis11/Barotrauma/assets/73229309/647e2c19-a711-4a9b-a8b3-a862fec9b6b8)

**Suggestion:**

Add a possibility for a "merchantlist" type of file in which you can list items, so you don't have to override invididual items.

![image](https://github.com/Regalis11/Barotrauma/assets/73229309/296f4902-6a81-4439-ad0f-eba6fbbc6c6f)

### Reproduction steps

a

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.2.7.0 (Winter Update hotfix)

### Which operating system did you encounter this bug on?

Windows

**Comment by WrillWasTaken** [comment]
> better to override 1 NPC than 50 items

Not really, because 1 NPC can only be overridden by one mod. Say you have 3 mods that want to add or adjust purchasable items in the same merchant, only one mod can have their items in the shop. This is a compatibility **nightmare**

> Modders resort to vending machines to circumvent the problem entirely.

Which mods do this? Could you give some examples?

Also, are vending machines not the same? They're fabricators that use money, so do you not have to define the recipe in the item itself anyway? Why not just add or adjust the Price element?
